### PR TITLE
Add custom metrics to Device Defender Demo

### PR DESCRIPTION
**Pull request by aggarw13**
Equivalent of PR #507 for release-candidate.

### CODE REVIEW

**Review by archigup — APPROVED** [review]

File: FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c [hunk 1]
```diff
@@ -35,18 +35,20 @@
  * with another MQTT library. This demo requires using the AWS IoT broker as
  * Device Defender is an AWS service.
  *
- * This demo connects to the AWS IoT broker and subscribes to the device
- * defender topics. It then collects metrics for the open ports and sockets on
- * the device using FreeRTOS+TCP, and generates a device defender report. The
+ * This demo subscribes to the device defender topics. It then collects metrics
```
Comment (gkwicker) [inline FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c:38]: Let's capitalize "Device Defender" wherever it appears in comments.

File: FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c [hunk 2]
```diff
@@ -463,7 +476,9 @@ static bool prvCollectDeviceMetrics( void )
 {
     bool xStatus = false;
     eMetricsCollectorStatus eMetricsCollectorStatus;
```
Comment (gkwicker) [inline FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/DemoTasks/DefenderDemoExample.c:478]: Variables with the same name as the type can be confusing – let's rename this variable, e.g.:

```c
eMetricsCollectorStatus eStatus;
```

File: FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/defender_config.h [hunk 3]
```diff
@@ -52,4 +52,44 @@
 
 /************ End of logging configuration ****************/
 
+/**
+ * AWS IoT Device Defender Service supports both long and short names for keys
+ * in the report sent by a device. For example,
+ *
+ * A device defender report using long key names:
```
Comment (gkwicker) [inline FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/defender_config.h:59]: See previous comment about capitalizing "Device Defender."

**Review by gkwicker — CHANGES_REQUESTED** [review]
See comments

**Comment by paulbartell** [comment]
I suggest using "git cherry-pick -x" so that it's clear what the upstream commit is.

File: FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/defender_config.h (hunk 3 above)
Comment (aggarw13) [reply FreeRTOS-Plus/Demo/AWS/Device_Defender_Windows_Simulator/Device_Defender_Demo/defender_config.h:59]: Have updated all instances to be "**D**~~d~~evice **D**~~d~~efender"

**Review by gkwicker — APPROVED** [review]

**Pull request merged**